### PR TITLE
Enable CoolProp in CI and fix recuperator convergence bug

### LINKED ISSUE #55: Fix clippy::pedantic warnings in CoolProp modules
## What

`cargo clippy --all-targets -- -W clippy::pedantic -D warnings` produces 28 warnings on main, all in the CoolProp module (`coolprop.rs`, `error.rs`, `ffi.rs`, `wrapper.rs`).

CI enforces pedantic, so these only pass because the CoolProp files are behind `cfg` feature gates and CI runs without CoolProp features.

## Breakdown

- 16 `doc_markdown` (unbackticked identifiers like `CoolProp`, `AbstractState`)
- 4 `redundant_closure`
- 4 `implicit_borrow_as_raw_pointer`
- 4 `casting usize to i64` (`cast_possible_wrap`)

## Why

Clean pedantic baseline so new warnings are immediately visible in local development with `--features coolprop-static`.

### LINKED ISSUE #58: Discretized recuperator fails to converge with CoolProp near CO₂ critical point
## What

The discretized recuperator solver fails to converge with CoolProp for sCO₂ conditions when using more than 1 segment. The error is:

```
recuperator solver failed to converge: invalid bracket: no sign change
```

This works fine with PerfectGas at the same conditions, and works with CoolProp at 1 segment. The issue appears with 5+ segments at conditions near CO₂'s critical point (31.1°C, 7.38 MPa).

## Conditions that reproduce it

Using the brayton cycle solver with these inputs:
- Compressor inlet: 32°C, 8 MPa (just above the critical point)
- Turbine inlet: 550°C, ~20 MPa (after pressure drops)
- Recuperator UA: 2000 kW/K
- Pressure drops: 2% each side
- Segments: 10 (fails), 5 (fails), 1 (works)

The same conditions with PerfectGas converge at all segment counts.

## Likely cause

Near the critical point, CO₂'s specific heat (cp) has a sharp spike — it can change by an order of magnitude over a few degrees. When the recuperator is discretized, individual segments can span regions where properties change dramatically. The bracketing solver assumes a smooth, monotonic residual function, but the cp spike can create non-monotonic behavior within a segment, breaking the sign-change assumption.

## Where to start

The solver is in `models::thermal::hx::discretized`. The bracketing logic that produces "invalid bracket: no sign change" is the entry point — look at what residual function it's evaluating and how the bracket bounds are chosen. The bracket may need to be widened, or the solver may need a fallback strategy when the initial bracket doesn't contain a sign change.

## Suggested tests

1. **Direct recuperator test with CoolProp at sCO₂ conditions.** Construct a `Recuperator` with `CoolProp<CarbonDioxide>`, 10 segments, and inlet states matching the conditions above (compressor outlet ~180°C at 20 MPa on the cold side, turbine outlet ~380°C at 8 MPa on the hot side). This should converge but currently doesn't.

2. **Sweep segment count.** Same conditions, iterate segments from 1 to 50. Identify where convergence first fails — this narrows down whether it's a specific segment hitting the cp spike or a cumulative numerical issue.

Both tests are CoolProp-specific — PerfectGas doesn't exhibit the property gradients that trigger this.

### PR DESCRIPTION
**Pull request by dyreby**
## What

Enable CoolProp in CI and fix a recuperator convergence bug that surfaced with it.

CI now runs clippy and tests with `--features coolprop-dylib`. The `coolprop-sys-linux-x86-64` dylib crate was already a dependency — we just weren't using it. This means CoolProp-gated code gets checked on every PR instead of only during local development.

Enabling CoolProp in CI required fixing the 28 existing `clippy::pedantic` warnings in the CoolProp modules (#55), which were previously hidden behind the feature gate.

With CoolProp tests running, a new recuperator regression test at sCO₂ conditions exposed #58: the discretized solver fails with "invalid bracket: no sign change" when enthalpy differences at a bracket endpoint are pure floating-point noise (~1e-10 J/kg). A relative tolerance (1e-12) on the enthalpy difference in the resolution step fixes it.

Closes #58
Closes #55

### CODE REVIEW

**Comment by dyreby** [comment]
CI was failing because `libCoolProp.so` has SONAME `libCoolProp.so.7`. The build script copies the `.so` into `OUT_DIR` and sets rpath, but the runtime linker resolves by SONAME — so it looks for `libCoolProp.so.7`, which didn't exist. Fixed by creating a soname symlink before running tests (5bb0cc2).

File: src/models/thermal/hx/discretized/core/solve/resolved.rs [hunk 1]
```diff
@@ -1,10 +1,22 @@
 //! Resolution of boundary conditions to determine heat exchanger endpoints.
 
+/// Relative tolerance for treating enthalpy differences as zero.
+///
+/// When the enthalpy change across a stream is smaller than this fraction
+/// of the inlet enthalpy, the heat transfer is treated as zero.
+/// This prevents floating-point noise in real-gas property evaluations
+/// from being misclassified as directional heat flow, which would cause
+/// spurious second-law violations in the solver.
+const ENTHALPY_RELATIVE_ZERO: f64 = 1e-12;
```
[the remark this answers is not in the document]
Comment (dyreby) [reply src/models/thermal/hx/discretized/core/solve/resolved.rs]: Good catch. The guard does work for the current case — sCO₂ puts h_in at ~10⁵ J/kg, so the threshold lands around 1e-7 J/kg, well above the ~1e-10 J/kg noise we were seeing. But you're right that it weakens as h_in approaches zero, and since enthalpy references are arbitrary there's no fully general fix. Added a doc comment in 5b9cf4f noting the limitation and pointing future readers here if they hit a similar issue with a different fluid.
